server/aws-lsp-codewhisperer/src/language-server/telemetryService.ts

```diff
@@ -0,0 +1,83 @@
+import { CodeWhispererServiceToken } from './codeWhispererService'
+import { CredentialsProvider, Telemetry } from '@aws/language-server-runtimes/server-interface'
+import { CodeWhispererSession } from './session/sessionManager'
+import {
+    SuggestionState,
+    UserTriggerDecisionEvent,
+    UserContext,
+    OptOutPreference,
+} from '../client/token/codewhispererbearertokenclient'
+import { getCompletionType } from './utils'
+
+export class TelemetryService extends CodeWhispererServiceToken {
+    private userContext!: UserContext
+    private optOutPreference!: OptOutPreference
+    private enableTelemetryEventsToDestination!: boolean
+    private toolkitTelemetry!: Telemetry
+
+    constructor(credentialsProvider: CredentialsProvider, additionalAwsConfig: AWS.ConfigurationOptions = {}) {
+        super(credentialsProvider, additionalAwsConfig)
+    }
+
+    public setToolkitTelemetry(toolkitTelemetry: Telemetry): void {
```

Comment:
I'm not sure for the purpose of this method from the name, it is needed at all, you can pass `telemetry` object in constructor. `UserContext` can be passed in constructor as well.

Reply:
I can pass telemetry in the constructor but userContext for let's say QServer is not defined within the same file but ChatController instead.